rad-exe/src/cli/args.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright © 2021 The Radicle Link Contributors
+//
+// This file is part of radicle-link, distributed under the GPLv3 with Radicle
+// Linking Exception. For full terms see the included LICENSE file.
+
+use std::env;
+
+use structopt::StructOpt;
+
+use librad::profile::ProfileId;
+
+/// `--rad-profile` command line name
+pub const RAD_PROFILE_ARG: &str = "--rad-profile";
+
+/// `--rad-quiet` command line name
+pub const RAD_QUIET_ARG: &str = "--rad-quiet";
+
+/// `--rad-verbose` command line name
+pub const RAD_VERBOSE_ARG: &str = "--rad-verbose";
+
+#[derive(Debug, StructOpt)]
+pub struct Args {
+    /// The profile identifier, if not given then the currently active profile
+    /// is used
+    #[structopt(long)]
+    pub rad_profile: Option<ProfileId>,
+
+    /// No output printed to stdout
+    #[structopt(long)]
+    pub rad_quiet: bool,
+
+    /// Use verbose output
+    #[structopt(long)]
+    pub rad_verbose: bool,
+
+    #[structopt(subcommand)]
+    pub command: Command,
+}
+
+#[derive(Debug, StructOpt)]
+pub enum Command {
+    /// This is just a stub, sshhhh
+    // TODO(finto): Fill in core commands
+    Profile,
+    #[structopt(external_subcommand)]
+    External(Vec<String>),
+}
+
+/// If an external subcommand is called, we sanitise the global arguments according to the rules defined in [RFC 698](https://github.com/radicle-dev/radicle-link/blob/master/docs/rfc/0698-cli-infrastructure.adoc#global-parameters).
+///
+/// The rules are summarised as:
+///   * The `rad` global _always_ takes precedence, e.g. `rad --rad-profile deaf
+///     xxx --rad-profile beef` will result in `deaf`
+///   * If multiple `rad` globals are given, it is an error, i.e. `rad
+///     --rad-profile deaf --rad-profile beef`
+///   * If multiple globals are given as part of the external command, the last
+///     one is take, e.g. `rad xxx --rad-profile deaf --rad-profile beef` will
+///     result in `beef`.
+///   * Command line arguments take precedence over environment variables, e.g.
+///     `RAD_PROFILE=deaf rad --rad-profile def` will result in `beef`
+///
+/// # Examples
+///
+/// ```text
+/// rad --rad-profile=deaf xxx --rad-profile=beef # deaf
+/// rad xxx --rad-profile=beef # beef
+/// rad --rad-profile=deaf --rad-profile=beef xxx # beef
+/// rad xxx --rad-profile=deaf --rad-profile=beef # beef
+/// rad --rad-profile=dead xxx --rad-profile=deaf --rad-profile=beef # dead
+/// ```
+pub fn sanitise_globals(mut args: Args) -> Args {
+    match &mut args.command {
+        Command::External(external) => {
+            sanitise_option(
+                RAD_PROFILE_ARG,
+                "RAD_PROFILE",
```

Comment:
Could be a `const`.

*ducks for cover*

Reply:
And/or bundle this one and the other `RAD_`s up in an env module.

Reply:
It's already defined here https://github.com/radicle-dev/radicle-link/blob/master/librad/src/profile.rs#L18 so I can make that `pub` :)

Reply:
`pub` it all.